src/EA.Weee.RequestHandlers/AatfReturn/GetSchemeRequestHandler.cs

```diff
@@ -0,0 +1,51 @@
+﻿namespace EA.Weee.RequestHandlers.AatfReturn
+{
+    using EA.Prsd.Core.Mediator;
+    using EA.Weee.Core.Scheme;
+    using EA.Weee.RequestHandlers.Security;
+    using EA.Weee.Requests.AatfReturn;
+    using Prsd.Core.Mapper;
+    using System;
+    using System.Collections.Generic;
+    using System.Threading.Tasks;
+    using Scheme = Domain.Scheme.Scheme;
+
+    public class GetSchemeRequestHandler : IRequestHandler<GetReturnScheme, List<SchemeData>>
+    {
+        private readonly IWeeeAuthorization authorization;
+        private readonly IReturnSchemeDataAccess returnSchemeDataAccess;
+        private readonly RequestHandlers.Scheme.IGetSchemesDataAccess schemeDataAccess;
+        private readonly IMapper mapper;
+  
+        public GetSchemeRequestHandler(IWeeeAuthorization authorization, IReturnSchemeDataAccess returnSchemeDataAccess, RequestHandlers.Scheme.IGetSchemesDataAccess schemeDataAccess, IMapper mapper)
+        {
+            this.authorization = authorization;
+            this.returnSchemeDataAccess = returnSchemeDataAccess;
+            this.schemeDataAccess = schemeDataAccess;
+            this.mapper = mapper;
+        }
+
+        public async Task<List<SchemeData>> HandleAsync(GetReturnScheme message)
+        {
+            authorization.EnsureCanAccessExternalArea();
+
+            List<Scheme> schemeList = new List<Scheme>();
+            List<SchemeData> schemeListData = new List<SchemeData>();
+            List<Guid> schemeIdList = await returnSchemeDataAccess.GetSelectedSchemesByReturnId(message.ReturnId);
+
+            foreach (var schemeId in schemeIdList)
+            {
```

Comment:
We don't need to loop here, alter the GetSelectedSchemesByReturnId to load the Scheme objects and return list ReturnScheme
  rather than loading the Id's and then each Scheme for the Id
        public async Task<List<ReturnScheme>> GetSelectedSchemesByReturnId(Guid returnId)
        {
            return await context.ReturnScheme
                    .Include(r => r.Scheme)
                .Where(now => now.ReturnId == returnId).Select(now => now.SchemeId).ToListAsync();
        }

Update the tests as this will break them.

Reply:
so have changed the code to:
schemeIdList = await returnSchemeDataAccess.GetSelectedSchemesByReturnId(message.ReturnId);
So schemeList needs to be converted to Scheme type? to use schemeList.AddRange(schemeIdList);
Is this what you meant?